docs/user-guides/best-practices.md

```diff
@@ -0,0 +1,165 @@
+
+### Laying Out Your Python Code
+
+Hera aims to keep orchestration code outside of your business logic, meaning you should write business logic in
+functions decorated with `@script`, and construct your Workflow separately, as seen throughout the examples. This might
+involve putting code in different submodules in your application, but overall you are free to write your Workflow how
+you want.
+
+An example directory structure for a project using `poetry` might be:
+
+```console
+.
+├── Dockerfile
+├── hera_scratch
+│   ├── __main__.py
+│   └── workflow.py
+├── Makefile
+├── poetry.lock
+├── pyproject.toml
+├── README.md
+└── requirements.txt
+```
+
+In this project, we use:
+
+* a `Dockerfile` to create an image for the runner script, using the packages in `requirements.txt` and a command
+  contained in the `Makefile`
+* a `__main__.py` file to run the Workflow from `workflow.py` on Argo Workflows
+* standard poetry/Python files
+
+See this layout in the [example Hera project repo](https://github.com/elliotgunton/hera-example-project)!
+
+### Steps vs DAGs
+
+* Use Steps for simple, sequential processing
+* Use DAGs to run as many tasks in parallel as possible, and if you want to only describe dependencies, not ordering
+
+### Iterating on a Workflow
+
+The developer process of iterating on a Workflow can be cumbersome as Argo does not offer a way to "dry run" your
+Workflow, and developers often get caught out by simple errors (such as unresolved variables).
+
+Hera should help reduce the number of iterations you need on the Workflow itself, but first you should create your
+business logic in isolation, and test it as normal Python (if you are using Python). The Workflow should then be written
+with the intention to test the plumbing between steps on a live Argo Workflows installation. You should have easily
+"mockable" business logic, in the sense that if your Workflow will process millions of rows of data, try running the
+Workflow with just a few rows at first. You can then iteratively build the Workflow from start to finish, testing the
+Workflow regularly on the live cluster.
+
+To summarise:
+
+1. Write business logic (and tests!)
+1. Add script decorators
+1. Write your Workflow
+1. Test as you go!
+
+## Workflows vs WorkflowTemplates vs ClusterWorkflowTemplates
+
+WorkflowTemplates are intended to be collections of templates that live in your Kubernetes namespace and used by other
+Workflows. ClusterWorkflowTemplates are the same but are accessible from all Kubernetes namespaces, so should perform
+common actions like email or Slack alerts, as they will be accessible by teams across your organisation. For brevity
+throughout the docs, when we refer to "WorkflowTemplates" we are also referring to "ClusterWorkflowTemplates".
+
+In Hera, you will usually be writing Workflows, unless you find a common pattern or template usage, which can then be
+extracted into a WorkflowTemplate.
+
+### Python Function Libraries vs WorkflowTemplates
```

Comment:
Not sure what is the conclusion or recommendation here. Would be nice to have a table comparing both and when to use what.